lib/migrate.js

```diff
@@ -0,0 +1,448 @@
+          indexesToAdd = _.merge(indexesToAdd, iAdd);
+        }
+
+        // function arrEqual(arr1, arr2) {
+        //   return _.isEqual(arr1.sort(), arr2.sort());
+        // }
+        // IMPROVE: use _.isEqual with customized compare function?
+        // index fields array is in format: [{afield: 'asc'}, {anotherfield: 'asc'}]
+        function arrEqual(arr1, arr2) {
+          assert(Array.isArray(arr1) && Array.isArray(arr2), 'make sure you pass in two Array!');
+          var notEqualMsg = util.inspect(arr1, 4) + ' is not equal to ' +
+        util.inspect(arr2, 4);
+          var isEqual = true;
+          for (var item in arr1) {
+            var i = _.findIndex(arr2, arr1[item]);
+            isEqual = isEqual && (i > -1);
+          }
+          return isEqual;
+        }
+      }
+    }
+    indexesToDrop = oldIndexes;
+    // Important, make sure `indexesToAdd` and `indexesToDrop` are object
+    result.indexesToAdd = indexesToAdd;
+    result.indexesToDrop = indexesToDrop;
+    return result;
+  };
+
+  /**
+ * Get all indexes of a model.
+ * 
+ * @param {String} model The model name
+ * @callback {Function} cb The callback function
+ * @param {Object} existingIndexes indexes in database that belongs to the model
+ * - existingIndexes format see jsdoc of `CouchDB.prototype.parseIndexes`
+ */
+  CouchDB.prototype.getModelIndexes = function(model, cb) {
+    debug('CouchDB.prototype.getModelIndexes: %j', model);
+    var self = this;
+    var mo = self.selectModel(model);
+    var dbName = mo.dbName;
+    self.getIndexes(dbName, function(err, result) {
+      if (err) return cb(err);
+      var ddocsOfModel = _.filter(result.indexes, isSameModel);
+      function isSameModel(item) {
+        var isSame = false;
+        if (item.ddoc !== null) {
+        // slice the '_design/'
+          var ddocName = item.ddoc.slice(8);
+          // need to be careful on the subString comparison based on 
+          // naming convertion, avoid one index belongs to two models
+          var modelIndexName = self.getIndexModelPrefix(mo) + '__' + model + '__' +
+          self.getIndexPropertyPrefix(mo);
+          if (ddocName.indexOf(modelIndexName) === 0)
+            isSame = true;
+        }
+        return isSame;
+      };
+
+      var existingIndexes = self.parseIndexes(ddocsOfModel);
+      cb(null, existingIndexes);
+    });
+  };
+
+  /**
+ * Parse the raw index object returned from database 
+ * to the format we need in connector
+ * Example:
+ * 
+ * raw index object:
+ * ```js
+ * {
+ *   ddoc: "_design/LBModel__User__LBIndex__name",
+ *   name: "name_index",
+ *   def: {
+ *     fields: [
+ *       {firstName: "asc"},
+ *       {lastName: "asc"}
+ *     ]
+ *   }
+ * }
+ * ```
+ * converts to the format: 
+ * {
+ *   name_index: {
+ *     ddoc: "_design/LBModel__User__LBIndex__name",
+ *     fields: [
+ *       {firstName: "asc"},
+ *       {lastName: "asc"}
+ *     ]
+ *   }
+ * }
+ * 
+ * @param {Object} ddocs Raw index object
```

Comment:
Suggestion: Can we rename the parameter, so that it matches the description? like `indices`? For me, `ddocs` means design docs